lib/git_fame/base.rb

```diff
@@ -185,8 +187,9 @@ def populate
           raise BranchNotFound.new("Does '#{@branch}' exist?")
         end
 
-        @files = execute("git ls-tree -r #{@branch} --name-only #{@include}").
-          split("\n")
+        command = "git ls-tree -r #{@branch} --name-only #{@include}"
+        command += " | grep \"\\.\\(#{@extensions.join("\\|")}\\)$\"" unless @extensions.empty?
```

Comment:
I originally had this as a regex check down in `remove_excluded_files`
```ruby
next unless path =~ /\.(#{@exclude.join("|")})\z/
```
But in the end thought it made more sense to filter them out from the start? Unfortunately I couldn't find a way to pass a regex directly to `ls-tree`, like you can with `ls-files`.